### ISSUE TITLE
sink-handlers-copy is redundent

### ISSUE THREAD

**Issue by tmccombs** [deadtrickster/cl-events#1]
The single-threaded sink contains two identical copies of the handlers list. I don't understand why this is necessary. If there is a good reason for it could you add comments explaining the reason?

**Comment by deadtrickster** [comment]
I think copy created and returned to prevent destructive modification. E.g. one can use RPLACA or something like that and this will break sink state. However looks like in that case copy should be created everytime handlers list returned. Thoughts?

**Comment by tmccombs** [comment]
Yes, if the goal is to prevent destructive modification, the sequence should be copied every time the handlers list is returned. Otherwise a client could destructively modify the copy, and sink-handlers-list would return the modified list. 

This would have performance implications as the list would be copied every time an event is fired. Maybe have a separate function that returns the original list for read-only usage, specifically iterating over the handlers when an event is fired.

**Comment by deadtrickster** [comment]
Looks like the job for iterators!

``` lisp
(let ((next-handler (sink-get-handlers-iterator)))
  (multiple-value-bind (handler more?) (next-handler)
     ... 
   ))
```

This allows us to abstract storage structure used by particular sink implementation.
Also definitely slower than just returning list.

Or maybe just leave lists and return them without any care.

**Comment by tmccombs** [comment]
Iterators seem like a good idea to me.

**Comment by deadtrickster** [comment]
unfortunately iterators do not play well with threaded sink. looks like returning fresh copy is a way to go here.
